File: HackerEarth/inscription/grid.py
def gcd(a,b):
	if b == 0:
		return a
	else:
		return gcd(b,a%b)

n,m = raw_input().split(" ")
n = int(n)
m = int(m)
s = ['']*502
for i in range(n):
	s[i] = raw_input()

ct = int(0)
for i in range(n-1):
	for j in range(i+1,n):
		z = int(0)
		for k in range(m):
			if (s[i][k] == '0' and s[j][k] == '0'):
				z+=1
		z-=1
		ct +=(z*(z+1))/2 

if ct == 0:
	print "1 1"
else:
	num1 = ct-1
	den1 = ct
	num2 = int(5)
	den2 = 6*ct	
	lcm = gcd(den1,den2)
	lcm = (den1*den2)/lcm

	num1*= lcm/den1
	num2*= lcm/den2
	num1 = num1+num2
	den1 = lcm
	lcm = gcd(num1, den1)
	num1/=lcm
	den1/=lcm
	print num1,den1

File: HackerEarth/inscription/grid.cpp
#include <bits/stdc++.h>
using namespace std;
int gcd(int a,int b)
{
	if(b==0) return a;
	else return gcd(b,a%b);
}
int main()
{
	int n,m; scanf("%d%d",&n,&m);
	string s[502];
	for(int i=0;i<n;i++) cin>>s[i];
	long long int ct=0;
	for(int i=0;i<n-1;i++)
	{
		long long int zero=0;
		for(int j=i+1;j<n;j++)
		{
			zero=0;
			for(int k=0;k<m;k++)
			{
				if(s[i][k] == '0' && s[j][k] == '0') zero++;
			}
			zero--;
			ct+=(zero*(zero+1))/2;
		}
	}
	//cout<<ct<<endl;
	long long int num1,num2,den1,den2;
	if(ct == 0)
	{
		printf("1 1\n");
	}
	else
	{
		num1=ct-1; den1=ct;
		num2 = 5; den2 = 6*ct;
		int lcm = gcd(num2,den2);
		num2/=lcm; den2/=lcm;

		lcm = gcd(den1,den2);
		lcm = (den1)/lcm;
		lcm *=den2;
		
		num1*= lcm/den1;
		num2*= lcm/den2;
		num1 = num1+num2;
		den1 = lcm;
		lcm = gcd(num1,den1);
		num1/=lcm; den1/=lcm;
		printf("%lld %lld\n",num1,den1);
	}
	
	return 0;
}